test/parallel/test-vm-property-not-on-sandbox.js

```diff
@@ -0,0 +1,63 @@
+'use strict';
+require('../common');
+const assert = require('assert');
+const vm = require('vm');
+
+// These, admittedly contrived, examples test two edge cases of the vm module.
+//
+// Both GetterCallback and QueryCallback
+// explicitly check the global_proxy() if a property is
+// not found on the sandbox. In the following tests, the explicit check
+// inside the callback yields different results than deferring the
+// check until after the callback. The check is deferred, if the
+// callbacks do not intercept, i.e., if args.GetReturnValue().Set() is
+// not called.
+
+{
+// Check that the SetterCallback explicitly calls GetRealNamedProperty()
+// on the global proxy if the property is not found on the sandbox.
+//
+// foo is not defined on the sandbox until we call CopyProperties().
+// In the GetterCallback, we do not find the property on the sandbox and
+// get the property from the global proxy. Since the return value is
+// the sandbox, we replace it by
+// the global_proxy to keep the correct identities.
+//
+// This test case is partially inspired by
+// https://github.com/nodejs/node/issues/855
+  const sandbox = {console};
+  sandbox.document = {defaultView: sandbox};
+  vm.createContext(sandbox);
+  const code = 'Object.defineProperty(' +
+               '  this, ' +
+               '  "foo", ' +
+               '  { get: function() {return document.defaultView} }' +
+               ');' +
+               'var result = foo === this;';
+
+  vm.runInContext(code, sandbox);
+  assert.strictEqual(sandbox.result, true);
+}
+
+{
+// Check that the QueryCallback explicitly calls GetRealNamedPropertyAttributes
+// on the global proxy if the property is not found on the sandbox.
+//
+// foo is not defined on the sandbox until we call CopyProperties().
+// In the QueryCallback, we do not find the property on the sandbox
+// and look up its PropertyAttributes on the global_proxy().
+// PropertyAttributes are always flattened to a value
+// descriptor.
+  const sandbox = {};
+  vm.createContext(sandbox);
+  const code = 'Object.defineProperty(' +
+               '  this, ' +
+               '  "foo", ' +
+               '  { get: function() {return 17} }' +
+               ');' +
+               'var desc = Object.getOwnPropertyDescriptor(this, "foo");' +
+               'var result = typeof desc.value === "number";';
```

Comment:
I don't follow why this would be the expected result instead of `typeof desc.value === 'undefined'` (and `typeof desc.get === 'function'`.)  Is this a test that better belongs in test/known_issues?